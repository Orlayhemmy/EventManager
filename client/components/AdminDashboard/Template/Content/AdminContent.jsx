import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect, Link } from 'react-router-dom';
import Centers from '../../../Common/Centers';
import Search from '../../../Common/Search';
import {
  getCenters,
  centerSelected,
  viewCenterSelected
} from '../../../../actions/centerActions';
import { searchValidation } from '../../../../shared/centerValidations';

/**
 * @description AdminPanelPage  component
 */
export class AdminPanelPage extends React.Component {
  /**
   * @memberof AdminDashMethod
   * @method render
   * @description it renders the component
   * @returns {void} the HTML of AdminDashMethod
   */
  constructor() {
    super();
    this.state = {
      counter: 0,
      location: '',
      facilities: '',
      capacity: '',
      capacityType: '',
      errors: {},
      btwValue: ''
    };
    this.showCenter = this.showCenter.bind(this);
    this.nextCenters = this.nextCenters.bind(this);
    this.onDelete = this.onDelete.bind(this);
    this.search = this.search.bind(this);
    this.onChange = this.onChange.bind(this);
    this.isValid = this.isValid.bind(this);
  }
  /**
   * @memberof AdminDashMethod
   * @method nextCenters
   * @description it fetches the next centers
   * @returns {void}
   * @param {object} e
   */
  nextCenters(e) {
    if (e.target.id === 'next') {
      this.setState({
        counter: this.state.counter + 1
      });
      this.props.getCenters(this.state, ++this.state.counter);
    } else {
      this.setState({
        counter: this.state.counter - 1
      });
      this.props.getCenters(this.state, --this.state.counter);
    }
  }
  /**
   * @memberof AdminDashMethod
   * @method showCenter
   * @description it fetches the details of the center to be viewed
   * @param {object} e
   * @returns {void}
   */
  showCenter(e) {
    this.props.viewCenterSelected(e.target.id);
  }

  /**
   * @memberof AdminDashMethod
   * @method onDelete
   * @description it fetches the details of the center to be deleted
   * @param {object} e
   * @returns {void}
   */
  onDelete(e) {
    const center = {
      centerId: e.target.id,
      centerName: e.target.parentNode.id
    };
    this.props.centerSelected(center);
  }

  /**
   * @memberof AdminDashMethod
   * @method search
   * @description it calls a search action
   * @param {object} e
   * @returns {void}
   */
  search(e) {
    e.preventDefault();
    this.setState({
      counter: 0
    });
    if (this.isValid()) {
      this.props.getCenters(this.state, this.state.counter);
    }
  }

  /**
   * @memberof AdminDashMethod
   * @method onChange
   * @description it sets user input to state
   * @param {object} e
   * @returns {object} state
   */
  onChange(e) {
    this.setState({
      [e.target.id]: e.target.value
    });
    const div = document.getElementById('btwValue');
    if (e.target.value === 'between' || e.target.id === 'btwValue') {
      div.hidden = false;
    } else {
      div.hidden = true;
    }
  }
  /**
   * @memberof AdminDashMethod
   * @method isValid
   * @description it calls validation action on user data
   * @returns {void} true or false
   */
  isValid() {
    const { errors, isValid } = searchValidation(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
  render() {
    //Check if user is logged in and is also an Admin
    if (!this.props.user.isAuth) {
      return <Redirect to="/" />;
    } else if (!this.props.user.user.isAdmin) {
      return <Redirect to="/dashboard" />;
    }
    return (
      <div className="container">
        <Search 
          search={this.search}
          criteria={this.state}
          onChange={this.onChange}
        />
        <Centers 
          // path={pathname}
          searchState={this.state}
          counter={this.state.counter}
          showCenter={this.showCenter}
          nextCenters={this.nextCenters}
          />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.auth
});

const propTypes = {
  centerSelected: PropTypes.func.isRequired,
  getCenters: PropTypes.func.isRequired,
  viewCenterSelected: PropTypes.func.isRequired
}

AdminPanelPage.propTypes = propTypes;

export default connect(mapStateToProps, {
  centerSelected,
  getCenters,
  viewCenterSelected
})(AdminPanelPage);
