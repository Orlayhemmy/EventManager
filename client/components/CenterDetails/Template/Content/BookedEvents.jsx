import React from 'react';

/**
 * @description ViewCenterDetails component
 */
export default class EventsList extends React.Component {
  render() {
    const { onClick } = this.props;
    const bookedEvents = this.props.eventState.map(event => {
      const {
        isApproved, bookedDate, id, eventTitle
      } = event;
      let eStatus;
      if (isApproved) {
        eStatus = <i id={id} className="fa fa-thumbs-up green" />;
      } else {
        eStatus = (
          <span
            onClick={onClick}
            data-toggle="modal"
            data-target="#deleteModal"
            id="approve"
          >
            <i id={id} className="fa fa-spinner main-color" />
          </span>
        );
      }
      return (
        <tr id={id} key={id}>
          <td>
            <span
              id={event.id}
              onClick={onClick}
              data-toggle="modal"
              data-target="#deleteModal"
            >
              {eventTitle}
            </span>
          </td>
          <td>{bookedDate.join(', ')}</td>
          <td>{eStatus}</td>
          <td>
            <span
              onClick={onClick}
              data-toggle="modal"
              data-target="#deleteModal"
              id="trash"
            >
              <i id={event.id} className="fa fa-trash trash" />
            </span>
          </td>
        </tr>
      );
    });
    return (
      <div id="event-list" className="col-lg-5">
        <div className="form-outer text-center d-flex align-items-center">
          <div className="form-inner">
            <strong className="logo text-primary">events scheduled</strong>
            <div>
              <table
                cellPadding="0"
                className="table table-responsive table-hover text-center"
              >
                <thead>
                  <tr>
                    <th>title</th>
                    <th>date</th>
                    <th>status</th>
                    <th />
                  </tr>
                </thead>
                <tbody>
                  {bookedEvents}
                  <tr>
                    <td>
                      <i className="fa fa-spinner main-color" />
                      <br />
                      <span>Pending</span>
                    </td>
                    <td />
                    <td />
                    <td>
                      <i className="fa fa-thumbs-up green" />
                      <br />
                      <span>Approved</span>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
