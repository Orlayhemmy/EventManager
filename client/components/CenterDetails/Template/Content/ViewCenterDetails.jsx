import React from 'react';

/**
 *  @description CenterDetailsContent form component
 */
export default class Details extends React.Component {
  render() {
    const {
      showHiddenDiv,
      centerState: {
        centerName,
        location,
        facilities,
        description,
        imageUrl,
        capacity,
        cost
      }
    } = this.props;
    return (
      <div id="centerDetails">
        <div className="imageUpload">
          <img className="img-fluid dropzone" src={imageUrl} />
        </div>
        <div className="media-body text-center mb-4 mt-4">
          <strong className="logo text-primary mb-2">{centerName}</strong>
          <h3 className="mt-2">location</h3>
          <p>{location}</p>
          <h3 className="mt-2">capacity</h3>
          <p>{capacity}</p>
          <h3 className="mt-2">facilities</h3>
          <p>{facilities}</p>
          <h3 className="mt-2">price</h3>
          <p>{cost}</p>
          <h3 className="mt-2">description</h3>
          <p>{description}</p>
        </div>
        ...{' '}
        <i
        id="edit-centerDetails"
          data-toggle-id="editCenterDetails"
          className="fa fa-pencil-alt main-color"
          onClick={showHiddenDiv}
        >
          edit
        </i>
      </div>
    );
  }
}
