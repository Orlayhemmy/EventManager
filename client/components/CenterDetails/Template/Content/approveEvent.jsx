import React from 'react';

export default class ApproveEvent extends React.Component {
  render() {
    const {
      onApprove,
      onChange,
      showHiddenDiv,
      onClick,
      state: { comment }
    } = this.props;
    return (
      <div>
        <span>Are you sure you want to approve this event?</span>
        <div>
          <i
            id="approve"
            className="fa fa-thumbs-up green"
            onClick={onApprove}
          />
          <i
            data-toggle-id="disapprove"
            className="fa fa-thumbs-down trash"
            onClick={showHiddenDiv}
          />
          <br />
          <span>
            <br />Approve
          </span>
          <span>
            <br />Disapprove
          </span>
        </div>
        <div id="disapprove" hidden>
          <p>
            {' '}
            Disapproved event will be deleted. Are you sure you want to
            disapprove event?
          </p>
          <div class="form-group">
            <textarea
              class="form-control"
              id="comment"
              onChange={onChange}
              placeholder="Give reasons for disapproving this event"
              value={comment}
            />
          </div>
          <i
            id="disapprove"
            className="fa fa-trash trash"
            onClick={onApprove}
          />
          <i
            data-toggle-id="disapprove"
            className="fa fa-close"
            onClick={showHiddenDiv}
          />
          <br />
          <span>
            <br />delete
          </span>
          <span>
            <br />cancel
          </span>
        </div>
      </div>
    );
  }
}
