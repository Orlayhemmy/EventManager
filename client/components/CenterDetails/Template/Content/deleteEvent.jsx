import React from 'react';

/**
 * @description Delete component
 */
export default class DeleteContent extends React.Component {
  render() {
    const {
      onApprove,
      onChange,
      state: { reason, suggestion }
    } = this.props;
    return (
      <div className="form-inner">
        <span className="help-block">Are sure you want to delete event?</span>
        <br />
        <br />
        <div class="form-group">
          <textarea
            class="form-control"
            id="reason"
            onChange={onChange}
            placeholder="Give reasons for disapproving this event"
            value={reason}
          />
        </div>
        <div class="form-group">
          <textarea
            class="form-control"
            id="suggestion"
            onChange={onChange}
            placeholder="Suggestions"
            value={suggestion}
          />
        </div>
        <i className="fa fa-trash red" id="disapprove" onClick={onApprove} />
        <i className="fa fa-save green" />
        <br />
        <span>
          <br />Yes
        </span>
        <span>
          <br />No
        </span>
      </div>
    );
  }
}
