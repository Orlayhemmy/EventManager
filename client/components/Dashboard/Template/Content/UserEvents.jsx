import React from 'react';
import _ from 'lodash';
import isEmpty from 'lodash/isEmpty';
import { Redirect, Link } from 'react-router-dom';

export default class Content extends React.Component {
  render() {
    const { status, events } = this.props.userEvent;
    let eventId, editEventId, eventBody, form, content, legend;
    if (isEmpty(events)) {
      content = (
        <div className="emptyEvent img-fluid text-center ml-2 mt-3 pt-2">
          <span>
            <p className="display-3">No Event Booked Yet</p>
          </span>
        </div>
      );
    } else {
      legend = (
        <div className="row center">
          <span className="legend">
            Approved Booking: <i className="fa fa-thumbs-up green" />
          </span>
          <span className="legend">
            Edit Booking: ...<i className="fa fa-pencil-alt main-color" />
          </span>
          <span className="legend">
            Delete Booking: <i className="fa fa-trash trash" />
          </span>
        </div>
      );
      const eventsArray = this.props.userEvent.events;
      console.log(eventsArray);
      content = _.map(eventsArray, (bookedEvent, index) => {
        const { imageUrl } = bookedEvent.Center;
        let eStatus;
        if (bookedEvent.isApproved) {
          eStatus = <i className="fa fa-thumbs-up green float-left" />;
        } else {
          eStatus = <i className="fa fa-spinner main-color float-left" />;
        }
        eventId = `eventDetails${index}`;
        editEventId = `eventDetails${index}`;
        form = `form${index}`;
        let dateBooked = `date${index}`;
        return (
          <div className="center" key={index}>
            <div key={eventId} className="text-center">
              <div className="card p-1 mb-3 mw">
                <div id={index} className="grid-view">
                  <span>{eStatus}</span>
                  <img className="img m-auto" src={imageUrl} />
                  <h2>
                    <span className="media-heading" id={bookedEvent.centerId}>
                      <Link
                        to="/modify-event"
                        id={bookedEvent.id}
                        onClick={this.props.onClick}
                      >
                        {bookedEvent.eventTitle}
                      </Link>
                    </span>
                  </h2>
                </div>
                {/* <div id={eventId} hidden>
                    <div className="media-body">
                      <h3>
                        <span>Date: </span> {bookedEvent.bookedDate}
                      </h3>
                      <h3>
                        <span>Center: </span> {centerName}
                      </h3>
                      <h3>
                        <span>Capacity: </span> {capacity}
                      </h3>
                      <h3>
                        <span>Location: </span> {location}
                      </h3>
                      <h3>
                        <span>facilities: </span> {facilities}
                      </h3>
                      <h3>
                        <span>Event description: </span> {bookedEvent.description}
                      </h3>
                    </div>
                  </div> */}
                <span>
                  <Link
                    to="/modify-event"
                    id={bookedEvent.id}
                    onClick={this.props.onClick}
                    className="float-left"
                  >
                    ...<i className="fa fa-pencil-alt" />
                  </Link>
                  <i
                    id={event.id}
                    className="fa fa-trash trash float-right"
                    onClick={this.props.onDelete}
                    data-toggle="modal"
                    data-target="#deleteModal"
                  />
                </span>
              </div>
            </div>
          </div>
        );
      });
    }
    return (
      <div className="wc">
        {legend}
        {this.props.counter > 0 ? (
          <div className="page-icon bounce">
            <i
              className="fa fa-chevron-up icon"
              id="previous"
              onClick={this.props.nextEvents}
            />
          </div>
        ) : (
          ''
        )}
        <div className="row">{content}</div>
        {events.length > 8 ? (
          <div className="page-icon bounce">
            <i
              className="fa fa-chevron-down icon"
              id="next"
              onClick={this.props.nextEvents}
            />
          </div>
        ) : (
          ''
        )}
      </div>
    );
  }
}
