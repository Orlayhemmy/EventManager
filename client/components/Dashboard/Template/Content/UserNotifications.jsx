import React from 'react';

/**
 * @description Notifications  component
 */
export default class Notifications extends React.Component {
  render() {
    const recentActivity = this.props.activities.map((activity, index) => {
      const creationDate = activity.createdAt
        .replace(/-/g, '/')
        .replace('Z', '')
        .replace('T', ' ')
        .slice(0, 16);
      return (
        <div className="row card p-1 mb-1" key={index}>
          <span>
            <p
              className="activity-font mb-0 p-1"
              onClick={this.props.onClick}
              id={activity.eventId}
            >
              {creationDate}
              <br />
              {activity.description}
            </p>
          </span>
        </div>
      );
    });
    return <div>{recentActivity}</div>;
  }
}
