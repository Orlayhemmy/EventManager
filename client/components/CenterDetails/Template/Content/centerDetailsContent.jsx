import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import swal from 'sweetalert2';
import {
  getCenterSelected,
  modifyCenter
} from '../../../../actions/centerActions';
import {
  getEventSelected,
  modifyCenterEvent,
  deleteCenterEvent,
  getCenterEvents,
  clearEventState
} from '../../../../actions/eventActions';
import Modal from '../../../Modal/Container/deleteModal';
import EditCenterForm from '../Form/editCenterForm';
import CenterDetails from './viewCenterDetails';
import { modifyCenterValidation } from '../../../../shared/centerValidations';
import uploadImage from '../../../../actions/imageAction';
import ApproveEvent from './approveEvent';
import BookedEvents from './bookedEvents';
import DeleteEvent from './deleteEvent';

/**
 * @description CenterDetailsContent form component
 */
export class CenterDetailsContent extends React.Component {
  /**
   * @memberof CenterDetailsContent
   * @description it creates an instance of centerdetailscontent
   * @param {object} props
   */
  constructor(props) {
    super(props);
    const {
      centerName,
      location,
      description,
      capacity,
      imageUrl,
      facilities,
      id,
      cost
    } = props.centerData.center;

    this.state = {
      centerName: centerName || '',
      location: location || '',
      facilities: facilities.join() || '',
      description: description || '',
      capacity: capacity || '',
      errors: '',
      event: '',
      imageUrl: imageUrl || '',
      id: id || '',
      cost: cost || '',
      decision: ''
    };
  }
  componentDidMount() {
    this.props.getCenterSelected();
    this.props.getCenterEvents();
  }
  /**
   * @memberof CenterDetailsContent
   * @method onChange
   * @description it sets user input to state
   * @param {object} e
   */
  onChange = e => {
    this.setState({
      [e.target.id]: e.target.value
    });
  };
  /**
   * @memberof CenterForm
   * @method showImage
   * @description it sets user input to state
   * @param {object} event
   */
  showImage = event => {
    if (event.target.files && event.target.files[0]) {
      const reader = new FileReader();
      this.state.imageData = event.target.files[0]; // eslint-disable-line
      reader.onload = e => {
        this.setState({ image: e.target.result });
      };
      reader.readAsDataURL(event.target.files[0]);
    }
  };
  /**
   * @memberof Profile
   * @method isValid
   * @description it calls validation action on user data
   * @returns true or false
   */
  isValid() {
    const { errors, isValid } = modifyCenterValidation(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
  /**
   * @memberof CenterDetailsContent
   * @method onSubmit
   * @description it calls the user signin action
   * @param {object} e
   * @returns {void}
   */
  onSubmit = e => {
    const {
      location, centerName, facilities, capacity, cost, id, image, description
    } = this.state;
    e.preventDefault();
    if (this.isValid()) {
      const formData = new FormData();
      formData.append('file', this.state.imageData);
      formData.append('upload_preset', 'u8asaoka');
      const data = {
        centerName,
        location,
        description,
        facilities,
        capacity,
        cost,
        id
      };
      if (this.initialState !== this.state) {
        if (this.initialState.image === image) {
          this.props.modifyCenter(this.state);
        } else {
          this.props.uploadImage(data, formData, 'modify-center');
        }
        this.showHiddenDiv(e);
      }
    }
  };
  /**
   * @memberof CenterDetailsContent
   * @method componentWillReceiveProps
   * @description it updates the state when new props are recieved
   * @param {object} nextProps
   * @returns {void}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.centerData !== this.props.center) {
      const {
        centerName,
        location,
        facilities,
        description,
        capacity,
        imageUrl,
        id,
        cost
      } = nextProps.centerData.center;
      this.setState(
        {
          centerName: centerName || '',
          location: location || '',
          facilities: facilities.join() || '',
          description: description || '',
          imageUrl: imageUrl || nextProps.centerData.url,
          capacity: capacity || '',
          id: id || '',
          cost: cost || ''
        },
        () => {
          this.initialState = this.state;
        }
      );
    }
  }
  /**
   * @memberof CenterDetailsContent
   * @method componentDidUpdate
   * @description it checks some conditions when component updates
   * @returns {void}
   */
  componentDidUpdate() {
    const { status } = this.props.eventState;
    if (status === 201 || status === 202 || status === 200) {
      swal('Changes Applied');
      $('#eventStatus').modal('hide');
      $('#deleteModal').modal('hide');
      this.props.clearEventState();
    }
    if (this.props.centerData.status === 202) {
      swal('Changes Applied');
    }
  }
  /**
   * @memberof CenterDetailsContent
   * @method onClick
   * @description it calls the get event action
   * @param {object} e
   * @returns {void}
   */
  onClick = e => {
    this.state.eventId = e.target.id;
    this.setState({
      decision: e.target.parentNode.id
    });
  };

  /**
   * @memberof CenterDetailsContent
   * @method onApprove
   * @description it calls an action when changes is made to events
   * @param {object} e
   * @returns {void}
   */
  onApprove = e => {
    const { eventTitle } = this.props.eventState.event;
    const { eventId } = this.state;
    const centerId = this.props.centerData.center.id;
    if (e.target.id === 'approve') {
      const data = {
        centerId,
        isApproved: true,
        id: eventId
      };
      this.props.modifyCenterEvent(data);
    } else {
      const data = {
        eventTitle,
        centerId,
        id: eventId,
        text: 'disapproved'
      };
      this.props.deleteCenterEvent(data);
      $('#deleteModal').modal('hide');
    }
  };
  /**
   * @memberof CenterDetailsContent
   * @method showHiddenDiv
   * @description it toggle's divs display
   * @param {object} e
   * @returns {void}
   */
  showHiddenDiv = e => {
    const id = e.target.dataset.toggleId;
    if (!id) return;
    const div = document.getElementById(id);
    div.hidden = !div.hidden;
    if (id === 'editCenterDetails') {
      const div2 = document.getElementById('centerDetails');
      if (!div.hidden) {
        div2.style.display = 'none';
        return;
      }
      div2.style.display = '';
    }
  };

  /**
   * @memberof CenterDetailsContent
   * @method render
   * @description it renders the component
   * @returns the HTML of centerdetails
   */
  render() {
    const { image, imageUrl, decision } = this.state;
    const { path } = this.props;
    const { events } = this.props.eventState;
    const content =
      decision !== undefined && this.state.decision === 'approve' ? (
        <ApproveEvent
          showHiddenDiv={this.showHiddenDiv}
          onChange={this.onChange}
          onApprove={this.onApprove}
          onClick={this.onClick}
          state={this.state}
        />
      ) : (
        <DeleteEvent
          state={this.state}
          showHiddenDiv={this.showHiddenDiv}
          onChange={this.onChange}
          onApprove={this.onApprove}
          onClick={this.onClick}
        />
      );

    return (
      <div id="center-event">
        <div className="container">
          <div className="row">
            <div className="col-lg-6 card bb text-center pt-4 pb-4">
              <CenterDetails
                centerState={this.state}
                showHiddenDiv={this.showHiddenDiv}
              />
              <EditCenterForm
                path={this.props.path}
                imageUrl={image || imageUrl}
                centerState={this.state}
                showImage={this.showImage}
                onSubmit={this.onSubmit}
                showHiddenDiv={this.showHiddenDiv}
                onChange={this.onChange}
              />
            </div>
            <BookedEvents onClick={this.onClick} eventState={events} />
            <Modal path={path} content={content} />
          </div>
        </div>
      </div>
    );
  }
}
const propTypes = {
  centerData: PropTypes.object.isRequired,
  eventState: PropTypes.object.isRequired,
  getEventSelected: PropTypes.func.isRequired,
  modifyCenterEvent: PropTypes.func.isRequired,
  deleteCenterEvent: PropTypes.func.isRequired,
  getCenterEvents: PropTypes.func.isRequired,
  getCenterSelected: PropTypes.func.isRequired,
  modifyCenter: PropTypes.func.isRequired,
  uploadImage: PropTypes.func.isRequired,
  clearEventState: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  centerData: state.center,
  eventState: state.event
});

CenterDetailsContent.propTypes = propTypes;

export default connect(
  mapStateToProps,
  {
    getEventSelected,
    modifyCenterEvent,
    deleteCenterEvent,
    getCenterEvents,
    getCenterSelected,
    modifyCenter,
    uploadImage,
    clearEventState
  }
)(CenterDetailsContent);
