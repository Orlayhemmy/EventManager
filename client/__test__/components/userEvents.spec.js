import React from 'react';
import Content from '../../components/Dashboard/Template/Content/UserEvents';
import { bookedEvent } from './MockData';

describe('Dashboard Component', () => {
  const props = {
    auth: {
      user: {
        isAdmin: true,
        id: 1
      },
      isAuth: true
    },
    userEvent: {
      status: '',
      message: '',
      events: []
    },
    location: {
      pathname: '/'
    },
    state: {
      eventId: 1,
      eventName: 'Pizza Party'
    },
    counter: 0,
    activity: {},
    nextEvents: jest.fn(),
    onSelect: jest.fn()
  };
  const wrapper = shallow(<Content {...props} />);

  it('should render component without error', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('check the length of the user events in the component', () => {
    wrapper.setProps({
      ...wrapper.props,
      userEvent: {
        status: '',
        message: '',
        events: bookedEvent
      },
      counter: 1
    });
    expect(wrapper.find('#booked-events').length).toEqual(10);
  });
});
