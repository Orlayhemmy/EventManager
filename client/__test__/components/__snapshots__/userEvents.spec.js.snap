// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component should render component without error 1`] = `
<div
  className="wc"
>
  <div
    className="row"
  >
    <div
      className="emptyEvent img-fluid text-center ml-2 mt-3 pt-2"
    >
      <span>
        <p
          className="display-3"
        >
          No Event Booked Yet
        </p>
      </span>
    </div>
  </div>
</div>
`;
